src/client/common/errors/errorUtils.ts

```diff
@@ -49,14 +33,34 @@ export function getErrorMessageFromPythonTraceback(traceback: string) {
     return lastLine.match(pythonErrorMessageRegExp) ? lastLine : undefined;
 }
 
-// eslint-disable-next-line @typescript-eslint/no-explicit-any
-type Constructor<T> = { new (...args: any[]): T };
-export function isErrorType<T>(error: Error, expectedType: Constructor<T>) {
-    if (error instanceof expectedType) {
-        return true;
+export function getLastFrameFromPythonTraceback(
+    traceback: string
+): { fileName: string; folderName: string; packageName: string } | undefined {
+    if (!traceback) {
+        return;
+    }
+    //             File "/Users/donjayamanne/miniconda3/envs/env3/lib/python3.7/site-packages/appnope/_nope.py", line 38, in C
+
+    const lastFrame = traceback
+        .split('\n')
+        .map((item) => item.trim().toLowerCase())
+        .filter((item) => item.length)
+        .reverse()
+        .find(
+            (line) =>
+                line.startsWith('file ') && line.includes(', line ') && line.includes('.pi') && line.includes('.pi')
+        );
+    if (!lastFrame) {
+        return;
     }
-    if (error instanceof WrappedError && error.originalException instanceof expectedType) {
-        return true;
+    const file = lastFrame.substring(0, lastFrame.lastIndexOf('.py')) + '.py';
+    const parts = file.replace(/\\/g, '/').split('/');
+    const indexOfSitePackages = parts.indexOf('site-packages');
+    let packageName =
+        indexOfSitePackages >= 0 && parts.length > indexOfSitePackages + 1 ? parts[indexOfSitePackages + 1] : '';
+    const reversedParts = file.replace(/\\/g, '/').split('/').reverse();
```

Comment:
Could just use parts and reverse that instead of replacing and splitting again, though I doubt it's a perf issue.